cub/test/catch2_test_device_scan_invalid.cu

```diff
@@ -0,0 +1,189 @@
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the NVIDIA CORPORATION nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL NVIDIA CORPORATION BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ ******************************************************************************/
+
+#include "insert_nested_NVTX_range_guard.h"
+
+#include <cub/device/device_scan.cuh>
+
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/iterator/transform_iterator.h>
+#include <thrust/iterator/zip_iterator.h>
+
+#include <cuda/cmath>
+#include <cuda/std/limits>
+
+#include <cstdint>
+
+#include "catch2_test_device_scan.cuh"
+#include "catch2_test_launch_helper.h"
+#include <c2h/catch2_test_helper.h>
+#include <c2h/custom_type.h>
+
+DECLARE_LAUNCH_WRAPPER(cub::DeviceScan::ExclusiveScan, device_exclusive_scan);
+DECLARE_LAUNCH_WRAPPER(cub::DeviceScan::InclusiveScan, device_inclusive_scan);
+DECLARE_LAUNCH_WRAPPER(cub::DeviceScan::InclusiveScanInit, device_inclusive_scan_with_init);
+
+// %PARAM% TEST_LAUNCH lid 0:1:2
+
+// Element type for scans
+template <typename OffsetT>
+struct Segment
+{
+  // Make sure that default constructed segments can not be merged
+  OffsetT begin = cuda::std::numeric_limits<OffsetT>::min();
+  OffsetT end   = cuda::std::numeric_limits<OffsetT>::max();
+};
+
+// Needed for final comparison with reference
+template <typename OffsetT>
+bool operator==(Segment<OffsetT> left, Segment<OffsetT> right)
+{
+  return left.begin == right.begin && left.end == right.end;
+}
+template <typename OffsetT>
+std::ostream& operator<<(std::ostream& os, const Segment<OffsetT>& seg)
+{
+  return os << "[ " << seg.begin << ", " << seg.end << " )";
+}
+
+// Needed for data input using fancy iterators
+template <typename OffsetT>
+struct Tuple2Seg
+{
+  using seg_t = Segment<OffsetT>;
+  __host__ __device__ seg_t operator()(thrust::tuple<OffsetT, OffsetT> interval)
+  {
+    return {.begin = thrust::get<0>(interval), .end = thrust::get<1>(interval)};
```

Comment:
important: I think designated initialization is C++20. We're still supporting C++17.
```suggestion
    return {.begin = thrust::get<0>(interval), .end = thrust::get<1>(interval)};
```